mds/management/commands/poll_providers.py

```diff
@@ -199,15 +200,30 @@ def get_body(self, provider, url):
     def process_status_changes(self, status_changes, provider):
         if self.verbosity > 1:
             self.stdout.write("Processing...")
+
+        # What is the latest event recorded in that series? (order is not asserted in the specs)
+        try:
+            last_start_time_polled = utils.from_mds_timestamp(
+                max(status_change["event_time"] for status_change in status_changes)
+            )
+        except (KeyError, ValueError):
+            # Data so bad there is no or nothing but invalid event times
+            logger.exception("No valid event_time found in status_changes series: %s", status_changes)
+            # How can we prevent from asking them again next time?
+            if provider.last_start_time_polled:
+                return provider.last_start_time_polled + datetime.timedelta(milliseconds=1)
+            # The provider really doesn't help!
+            return timezone.now()
```

Comment:
Oh OK the previous comment was written before I saw this addition. I am still a bit confused. It will be filled with "now" only the first time we poll ?

Reply:
Yes, when we're polling a new provider for the first time and its endpoint is clearly not MDS compliant. On the next try, we'll only ask event that are new from the last attempt.